enterprise_catalog/apps/api/v2/tests/test_enterprise_catalog_views.py

```diff
@@ -0,0 +1,159 @@
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_contains_content_items_implicit_access(self):
+        """
+        Verify the contains_content_items endpoint responds with 200 OK for
+        user with implicit JWT access
+        """
+        self.remove_role_assignments()
+        url = self._get_contains_content_base_url() + '?program_uuids=fakeX'
+        self.assert_correct_contains_response(url, False)
+
+    def test_contains_content_items_no_params(self):
+        """
+        Verify the contains_content_items endpoint errors if no parameters are provided
+        """
+        response = self.client.get(self._get_contains_content_base_url())
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_contains_content_items_not_in_catalogs(self):
+        """
+        Verify the contains_content_items endpoint returns False if the content is not in any associated catalog
+        """
+        self.add_metadata_to_catalog(self.enterprise_catalog, [ContentMetadataFactory()])
+
+        url = self._get_contains_content_base_url() + '?program_uuids=this-is-not-the-uuid-youre-looking-for'
+        self.assert_correct_contains_response(url, False)
+
+    def test_contains_content_items_in_catalogs(self):
+        """
+        Verify the contains_content_items endpoint returns True if the content is in any associated catalog
+        """
+        content_key = 'fake-key+101x'
+        relevant_content = ContentMetadataFactory(content_key=content_key)
+        self.add_metadata_to_catalog(self.enterprise_catalog, [relevant_content])
+
+        url = self._get_contains_content_base_url() + '?course_run_ids=' + content_key
+        self.assert_correct_contains_response(url, True)
+
+    def _create_restricted_course_and_run(self, catalog):
+        """
+        Helper to setup restricted course and run.
+        """
+        content_one = ContentMetadataFactory(content_key='org+key1', content_type=COURSE)
+        restricted_course = RestrictedCourseMetadataFactory.create(
+            content_key='org+key1',
+            content_type=COURSE,
+            unrestricted_parent=content_one,
+            catalog_query=catalog.catalog_query,
+        )
+        restricted_run = ContentMetadataFactory.create(
+            content_key='course-v1:org+key1+restrictedrun',
+            content_type=COURSE_RUN,
+        )
+        restricted_course.restricted_run_allowed_for_restricted_course.set(
+            [restricted_run], clear=True,
+        )
+        return content_one, restricted_course, restricted_run
+
+    def test_contains_catalog_key_restricted_runs_allowed(self):
+        """
+        Tests that a catalog is considered to contain a restricted run.
+        """
+        content_one, _, restricted_run = self._create_restricted_course_and_run(self.enterprise_catalog)
+
+        self.add_metadata_to_catalog(self.enterprise_catalog, [content_one, restricted_run])
```

Comment:
For this test to be fully rigorous (and accurate to what prod data looks like), I think we do not want to add `restricted_runs` to the catalog here.
```suggestion
        self.add_metadata_to_catalog(self.enterprise_catalog, [content_one])
```

It should theoretically be sufficient (and what we really want to test) that restricted run is related via `RestrictedCourseMetadata` which was set up above.

Reply:
I just tested this change and it causes the result to flip to False. 🤔

Reply:
Same, but I think I'm missing a test setup step of defining the `restricted_runs_allowed` for the test catalog's query.

Reply:
I think for this entire v2 test class, we're probably missing a step to modify the catalog query's content filter to include a top-level "restricted_runs_allowed" key.  Without that json key, some of the restricted runs logic gets short-circuited in the name of performance.